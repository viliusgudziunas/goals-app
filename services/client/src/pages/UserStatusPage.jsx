import React, { useContext } from 'react';

import UserStatus from '../components/UserStatus';
import NotAllowed from '../components/NotAllowed';
import { GlobalContext } from '../context/GlobalState';

const UserStatusPage = () => {
  const { authenticated } = useContext(GlobalContext);

  if (authenticated) {
    return <UserStatus />;
  }

  return <NotAllowed />;
};

export default UserStatusPage;
