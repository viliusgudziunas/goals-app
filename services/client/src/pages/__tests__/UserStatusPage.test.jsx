import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';

import UserStatusPage from '../UserStatusPage';

describe('<UserStatusPage />', () => {
  const wrapper = shallow(
    <Router location='/'>
      <UserStatusPage />
    </Router>
  );
  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe('When not authenticated, <UserStatusPage />', () => {
  const wrapper = shallow(<UserStatusPage />);
  it('should render a <NotAllowed />', () => {
    expect(wrapper.find('NotAllowed').length).toBe(1);
  });
});

// describe('When authenticated, <UserStatusPage />', () => {
//   const wrapper = shallow(<UserStatusPage />);
//   it('should render a <UserStatus />', () => {
//     expect(wrapper.find('UserStatus').length).toBe(1);
//   });
// });
