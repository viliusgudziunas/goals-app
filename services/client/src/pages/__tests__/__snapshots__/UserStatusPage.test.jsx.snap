// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserStatusPage /> should match the snapshot 1`] = `"<div class=\\"container\\"><br/><h1>Not Allowed</h1><hr/><p>You must be logged in to view this. Click <a href=\\"/login\\">here</a> to log back in.</p></div>"`;
